step-capstone/src/components/Finalized.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import {useEffect, useState, useRef} from 'react';
+import ReactDOM from 'react-dom';
+import TravelObject from './TravelObject';
+import TimeLine from './TimeLine';
+export default function Finalized(props) {
+    const hashMap = new Map();
+    let prevDiv = useRef([]);
+    const [displayDate, setDisplayDate] = useState(props.startDate);
+
+    const handleDisplayDateChange = (newDisplayDate) => {
+        setDisplayDate(newDisplayDate);
+    }
+
+    const sameDate = (timeA, timeB) => {
+      return (timeA.getDate() === timeB.getDate() && timeA.getMonth() === timeB.getMonth() && timeA.getFullYear() === timeB.getFullYear());
+    }
+    const pushItemToArrayInHashMap = (item, dateKey) => {
+        if (!hashMap.has(dateKey)) {
+            hashMap.set(dateKey, [item]);
+        }
+        else {
+            let array = hashMap.get(dateKey);
+            array.push(item);
+            hashMap.set(dateKey, array);
```

Comment:
you probably don't need to reset the array into the hashmap since it's still referencing the same array

Reply:
you're right. Abundant line of code.